plugins/obs-qsv11/common_utils.h

```diff
@@ -162,10 +162,12 @@ int GetFreeTaskIndex(Task *pTaskPool, mfxU16 nPoolSize);
 mfxStatus Initialize(mfxVersion ver, mfxSession *pSession,
 		     mfxFrameAllocator *pmfxAllocator, mfxHDL *deviceHandle,
 		     bool bCreateSharedHandles, bool dx9hack,
-		     enum qsv_codec codec); //vpl change
+		     enum qsv_codec codec, void **data); //vpl change
 
-// Release resources (device/display)
+// Release global shared resources  (device/display)
 void Release();
+// Release per session resources
+void Release2(void *data);
```

Comment:
I kind of dislike the `Release2` naming here because it implies that it replaces `Release` (this is how we have historically named other replacement functions that deprecate an older version). I'd prefer something a name that better reflects its purpose. Maybe `ReleaseSessionData`?

Reply:
It should replace Release, the current usage of globals is kind of insane.

Reply:
Hmm. That seems like this is an intermediate fix and that the more complete fix would be to refactor the Release calls/definition.

Reply:
Not fixing windows is indeed an intermediate state, but whether anyone will ever fix windows to avoid globals is a question.

Reply:
Renamed to avoid confusion